### PR TITLE
Improved interpolation of position and velocity

### PR DESCRIPTION
**Pull request by lsav**
Swap interpolator to do (by default) quadratic interpolation. We could change it to cubic for better accuracy.

This uses scipy as a backend.

### CODE REVIEW

File: kaos/algorithm/interpolator.py [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
+        self.segment_times = {}  # segment_id : list of times
+        self.segment_positions = {}  # segment_id : list of positions
+        self.segment_velocities = {}  # segment_id : list of velocities
+
+    @staticmethod
+    def vector_interp(times, vecs, new_times, kind):
+        """Interpolate vectors for arbitrary timestamps, based on known timestamps.
+
+        Args:
+            times: A 1-D array containing timestamps of the input data.
```
Comment (ZeyadTamimi) [inline kaos/algorithm/interpolator.py]: NIT:
```times (array): ```

File: kaos/algorithm/interpolator.py [hunk 2]
```diff
@@ -1,16 +1,44 @@
...
+        Return:
+            numpy array of interpolated points of the same length as targets, where ret[i]
+            is the interpolated result for targets[i].
+        """
+        # interpolate position, then velocity
+        approx = interpolate.interp1d(times, vecs, kind=kind, axis=0)
+        return approx(np.array(new_times))
 
     @staticmethod
     def linear_interp(platform_id, timestamp):
```
[the remark this answers is not in the document]
Comment (ZeyadTamimi) [reply kaos/algorithm/interpolator.py]: Keep for the moment but add a comment saying its less accurate or something. I also guess that the changes to linear interp dont affect us right?

File: kaos/algorithm/interpolator.py [hunk 3]
```diff
@@ -30,39 +58,69 @@ def linear_interp(platform_id, timestamp):
...
-            tvel[i] = np.interp(x=timestamp, xp=xp, fp=vel[:, i])
+        # do the interpolation
+        pos = Interpolator.vector_interp(times, positions, [timestamp], kind="linear")[0]
+        vel = Interpolator.vector_interp(times, velocities, [timestamp], kind="linear")[0]
 
-        return tuple(tpos), tuple(tvel)
+        return tuple(pos), tuple(vel)
 
-    def interpolate(self, timestamp):
+    def interpolate(self, timestamp, kind="quadratic"):
```
Comment (ZeyadTamimi) [inline kaos/algorithm/interpolator.py]: I dont think you should add this if its not used!

Comment (lsav) [reply kaos/algorithm/interpolator.py]: We may want to tweak the kind in the future. For example, "cubic" would be more accurate (but more computationally intensive). Having it as a default parameter makes it easy to test -- don't have to modify the source to try out a new setting.

File: kaos/algorithm/interpolator.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lsav) [reply kaos/algorithm/interpolator.py]: 👍

**Pull request merged**